src/collapse/perf.rs

```diff
@@ -396,16 +502,46 @@ impl Folder {
         self.stack.clear();
     }
 
-    fn finish<W: Write>(&self, mut writer: W) -> io::Result<()> {
-        let mut keys: Vec<_> = self.occurrences.iter().collect();
-        keys.sort();
-        for (key, value) in keys {
-            writeln!(writer, "{} {}", key, value)?;
-        }
-        Ok(())
+    /// The closure returned by this static method is what runs upfront when multiple cores are used
+    /// in order to identify how the input data can be cut up into multiple pieces in order to be
+    /// spread accross the multiple cores. It should return a vector with the locations (byte
+    /// indices) of the start of each stack. See `crate::collapse::Input::new`.
+    fn identify_stack_locations<'a>(
```

Comment:
It's quite sad that we end up repeating the parsing from above here. I wonder if there's a way to avoid doing that...